builder/vsphere/common/step_export.go

```diff
@@ -0,0 +1,295 @@
+		state.Put("error", err)
+		return multistep.ActionHalt
+	}
+
+	u := lease.StartUpdater(ctx, info)
+	defer u.Done()
+
+	cdp := types.OvfCreateDescriptorParams{
+		Name: s.Name,
+	}
+
+	m := vm.NewOvfManager()
+
+	if len(s.Options) > 0 {
+		exportOptions, err := vm.GetOvfExportOptions(m)
+		if err != nil {
+			state.Put("error", err)
+			return multistep.ActionHalt
+		}
+		var unknown []string
+		for _, option := range s.Options {
+			found := false
+			for _, exportOpt := range exportOptions {
+				if exportOpt.Option == option {
+					found = true
+					break
+				}
+			}
+			if !found {
+				unknown = append(unknown, option)
+			}
+			cdp.ExportOption = append(cdp.ExportOption, option)
+		}
+
+		// only printing error message because the unknown options are just ignored by vcenter
+		if len(unknown) > 0 {
+			ui.Error(fmt.Sprintf("unknown export options %s", strings.Join(unknown, ",")))
+		}
+	}
+
+	for _, i := range info.Items {
+		if !s.include(&i) {
+			continue
+		}
+
+		if !strings.HasPrefix(i.Path, s.Name) {
+			i.Path = s.Name + "-" + i.Path
+		}
+
+		err = s.Download(ctx, lease, i)
+		if err != nil {
+			state.Put("error", err)
+			return multistep.ActionHalt
+		}
+
+		ui.Message("exporting file: " + i.File().Path)
+		cdp.OvfFiles = append(cdp.OvfFiles, i.File())
+	}
+
+	if err = lease.Complete(ctx); err != nil {
+		state.Put("error", errors.Wrap(err, "unable to complete lease"))
+		return multistep.ActionHalt
+	}
+
+	desc, err := vm.CreateDescriptor(m, cdp)
+	if err != nil {
+		state.Put("error", errors.Wrap(err, "unable to create descriptor"))
+		return multistep.ActionHalt
+	}
+
+	target := getTarget(s.OutputDir, s.Name)
+	file, err := os.Create(target)
+	if err != nil {
+		state.Put("error", errors.Wrap(err, "unable to create file: "+target))
+		return multistep.ActionHalt
+	}
+
+	var w io.Writer = file
+	h, ok := s.newHash()
+	if ok {
+		w = io.MultiWriter(file, h)
+	}
+
+	_, err = io.WriteString(w, desc.OvfDescriptor)
+	if err != nil {
+		state.Put("error", errors.Wrap(err, "unable to write descriptor"))
+		return multistep.ActionHalt
+	}
+
+	if err = file.Close(); err != nil {
+		state.Put("error", errors.Wrap(err, "unable to close descriptor"))
+		return multistep.ActionHalt
+	}
+
+	if s.Sha == 0 {
+		// manifest does not need to be created, return
+		return multistep.ActionContinue
+	}
+
+	s.addHash(filepath.Base(target), h)
+
+	file, err = os.Create(filepath.Join(s.OutputDir, s.Name+".mf"))
+	if err != nil {
+		state.Put("error", errors.Wrap(err, "unable to create manifest"))
+		return multistep.ActionHalt
+	}
+
+	_, err = io.Copy(file, &s.mf)
+	if err != nil {
+		state.Put("error", errors.Wrap(err, "unable to write manifest"))
+		return multistep.ActionHalt
+	}
+
+	err = file.Close()
+	if err != nil {
+		state.Put("error", errors.Wrap(err, "unable to close file"))
+		return multistep.ActionHalt
+	}
+
+	return multistep.ActionContinue
+}
+
+func (s *StepExport) include(item *nfc.FileItem) bool {
+	if s.Images {
```

Comment:
Could you explain what I should expect when I set `"images": true`? I expected to see an iso and img file along side the other files but did not. 

Testing Results: 
```
"Export": {
  "sha": 1,
  "force": true,
  "output_directory": "output_vsphere_ovf",
  "images": true
}
```
```
⇶ cd output_vsphere_ovf
⇶  tree
.
├── example-ubuntu-disk-0.vmdk
├── example-ubuntu.mf
└── example-ubuntu.ovf

0 directories, 3 files

⇶  sha1sum -c example-ubuntu.mf 
example-ubuntu-disk-0.vmdk: OK
example-ubuntu.ovf: OK

```

Reply:
I think the images: true is related to if you have any attached floppy images and isos. The vmware documentation seems somewhat sparse which is why I just included what they had.